### PR TITLE
(1823) Make some changes to how budgets work in the application

### PR DESCRIPTION
**Pull request by pezholio**
This makes the following changes to budgets: 

* Transfers are removed as budget types
* `External non Official Development Assistance` removed as a budget type
* `External Official Development Assistance` budget type changed to `Other official development assistance`
*  The Direct (Newton Fund) and Direct (Global Challenges Research Fund) types have been replaced with one `Direct` budget type.

⚠️ There is also a data migration to be run once this code hits production ⚠️ 

### Before

image

### After

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Make some changes to how Budgets work in the application](https://trello.com/c/Kn1pKEl4/1823-make-some-changes-to-how-budgets-work-in-the-application)

**Review by CristinaRO — CHANGES_REQUESTED** [review]
Given the recent issues we've had with other enum radio button types not being preserved on edit, I'd feel more confident if "user can edit a budget" was also testing that editing an external budget shows the existing selection of an external budget and the providing org.

File: db/schema.rb [hunk 1]
```diff
@@ -43,17 +43,16 @@
     t.boolean "call_present"
     t.date "call_open_date"
     t.date "call_close_date"
-    t.string "intended_beneficiaries", array: true
```
[the remark this answers is not in the document]
Comment (pezholio) [reply db/schema.rb:46]: Yeah, sorry. I'm normally so careful with this!

**Review by CristinaRO — APPROVED** [review]
Thank you!

**Pull request merged**